File: 자프링-GraphQL-실습/src/main/java/com/junyss/graphqltest/common/util/ObjectUtil.java
package com.junyss.graphqltest.common.util;

import org.springframework.data.domain.Page;
import org.springframework.stereotype.Component;

import lombok.AccessLevel;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor(access = AccessLevel.PRIVATE)
@Component
public class ObjectUtil {

	public static <T> Boolean checkObjectExistence(Page<T> targetPageObject) {
		return targetPageObject.getTotalElements() == 0 && !targetPageObject.hasContent();
	}
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/common/util/ObjectUtil.kt
package com.junyharangstudy.kotlingraphqltest.common.util

import org.springframework.data.domain.Page
import org.springframework.stereotype.Component

@Component
class ObjectUtil {
    companion object {
        fun <T> checkObjectExistence(targetPageObject: Page<T>): Boolean {
            return targetPageObject.totalElements.toInt() == 0 && !targetPageObject.hasContent()
        }
    }
}